.github/workflows/build_and_deploy.yml

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2022 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+name: Build, test and deploy docker images
+
+# The builds are triggered with pushes on the 'main' branch or with a pull
+# request against the 'main' branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  base_version: 1
```

Comment:
Also please update this to `2` already now so that we see that the HIP build works correctly even if the base image isn't on dockerhub.